lib/matplotlib/axes/_base.py

```diff
@@ -2557,7 +2557,16 @@ def handle_single_axis(scale, autoscaleon, shared_axes, interval,
                 dl.extend(y_finite)
 
             bb = mtransforms.BboxBase.union(dl)
-            x0, x1 = getattr(bb, interval)
+            # Issue 18137
```

Comment:
This code looks *almost* the same as the code above except just written a different way.  Is it not possible to combine them?

Reply:
Line 2559 computes a bounding box, bb, that is used later in the function. I chose to keep this part of code to avoid introducing a new bug.

Reply:
I will see if it is possible to combine them to avoid iterating through the dl array.

Reply:
I agree with @jklymak this should be fixed by sorting out why the code in L2544-L2557 are not doing what we think they should (I think that the `all` in L2544 should be changed to `any`?) rather than adding another layer of checking here.